http/e2e/pkg/runner.go

```diff
@@ -88,21 +88,21 @@ func Run(cfg Config) error {
 			requestBody:        "maliciouspayload",
 			expectedStatusCode: 403,
 		},
-		{
-			name:               "Denied request with a malicious response header",
-			requestURL:         proxyURL + "/response-headers?pass=leak",
-			requestMethod:      "GET",
-			expectedStatusCode: 403,
-		},
-		{
-			name:               "Denied request with a malicious response body",
-			requestURL:         echoProxiedURL,
-			requestMethod:      "POST",
-			requestHeaders:     map[string]string{"Content-Type": "application/x-www-form-urlencoded"},
-			requestBody:        "responsebodycode",
-			expectedEmptyBody:  true,
-			expectedStatusCode: 403,
-		},
+		// {
```

Comment:
Tests that lead to `io.ErrShortWrite`

Reply:
Fixed implementing the proxy not as NewSingleHostReverseProxy but rather emulating it with a mux